File: epi_judge_java/epi/TreeLevelOrder.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.List;
import java.util.Queue;

public class TreeLevelOrder {

    @EpiTest(testDataFile = "tree_level_order.tsv")
    public static List<List<Integer>> binaryTreeDepthOrder(BinaryTreeNode<Integer> tree) {
        List<List<Integer>> ans = new ArrayList<>();
        if (tree == null) return ans;
        Queue<BinaryTreeNode<Integer>> q = new ArrayDeque<>();
        q.offer(tree);
        while (!q.isEmpty()) {
            int size = q.size();
            List<Integer> level = new ArrayList<>();
            for (int i = 0; i < size; i++) {
                var polled = q.poll();
                level.add(polled.data);
                if (polled.left != null) q.offer(polled.left);
                if (polled.right != null) q.offer(polled.right);
            }
            ans.add(level);
        }
        return ans;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "TreeLevelOrder.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/tree_level_order.cc
#include <memory>
#include <vector>

#include "binary_tree_node.h"
#include "test_framework/generic_test.h"

using namespace std;

vector<vector<int>> BinaryTreeDepthOrder(const unique_ptr<BinaryTreeNode<int>> &tree) {
    if (!tree) return {};
    queue<BinaryTreeNode<int> *> q({tree.get()});
    vector<vector<int>> result;

    while (!q.empty()) {
        size_t size = q.size();
        vector<int> level;

        for (size_t i{0}; i < size; i++) {
            auto popped = q.front();
            q.pop();

            if (popped->left) q.push(popped->left.get());
            if (popped->right) q.push(popped->right.get());

            level.emplace_back(popped->data);
        }

        result.emplace_back(level);
    }

    return result;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"tree"};
    return GenericTestMain(args, "tree_level_order.cc", "tree_level_order.tsv",
                           &BinaryTreeDepthOrder, DefaultComparator{},
                           param_names);
}
